File: 2026/07-July/09-Thursday/solution.cs
public class Solution {
    public bool[] PathExistenceQueries(int n, int[] nums, int maxDiff, int[][] queries) {
        int[] g = new int[n];
        int cnt = 0;
        
        for (int i = 1; i < n; i++) {
            if (nums[i] - nums[i - 1] > maxDiff) {
                cnt++;
            }
            g[i] = cnt;
        }
        
        bool[] ans = new bool[queries.Length];
        for (int i = 0; i < queries.Length; i++) {
            ans[i] = g[queries[i][0]] == g[queries[i][1]];
        }
        
        return ans;
    }
}

File: 2026/07-July/09-Thursday/solution.py
class Solution:
    def pathExistenceQueries(self, n: int, nums: list[int], maxDiff: int, queries: list[list[int]]) -> list[bool]:
        # g[i] will store the component index of node i
        g = [0] * n
        cnt = 0
        
        # Identify connected components based on sequential breaks
        for i in range(1, n):
            if nums[i] - nums[i - 1] > maxDiff:
                cnt += 1
            g[i] = cnt
            
        # For each query, true if both nodes are in the same component
        return [g[u] == g[v] for u, v in queries]
